staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/list_work_estimator.go

```diff
@@ -50,6 +50,15 @@ func (e *listWorkEstimator) estimate(r *http.Request, flowSchemaName, priorityLe
 		return WorkEstimate{InitialSeats: e.config.MaximumSeats}
 	}
 
+	if requestInfo.Name != "" {
+		// Requests with metadata.name specified are usually executed as get
+		// requests in storage layer so their width should be 1.
+		// Example of such list requests:
+		// /apis/certificates.k8s.io/v1/certificatesigningrequests?fieldSelector=metadata.name%3Dcsr-xxs4m
+		// /api/v1/namespaces/test/configmaps?fieldSelector=metadata.name%3Dbig-deployment-1&limit=500&resourceVersion=0
```

Comment:
question: is the cost same as a regular GET on etcd side, i see `fieldSelector` in the request URI. I am not very familiar with the storage logic, just wondering whether it will be as expensive as a regular LIST on the etcd side.

Reply:
I will look into this.

Reply:
cf #85445

Reply:
Yes - it's the same.
This is the part that is crucial:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L362-L364


Thanks for fixing this @marseel !